### PR TITLE
Change KiCad Board language to KiCad Legacy Layout

### PR DESCRIPTION
**Pull request by seppestas**
KiCad .brd files and .kicad_pcb files have the same purpose, they are both source files for PCB layouts. Having one of the file types named "KiCad Board" and the other one "KiCad Layout" can cause confusion since it implies they are not the same thing.

The [.brd files use the old, legacy layout format](http://kicad-pcb.org/help/file-formats/#_native_file_formats) that is [not actively used anymore](https://github.com/search?utf8=%E2%9C%93&q=language%3A%22KiCad+Board%22&type=Repositories&ref=advsearch&l=KiCad+Board&l=). Having it come before the KiCad Layout language in the Language Selection list and not having it flagged as legacy can cause people to select it when searching for KiCad layout files:

screen shot 2017-08-29 at 10 27 56

See #3784 for the ongoing discussion about the KiCad file formats. This PR builds upon the changes introduced by #3743.

### CODE REVIEW

**Comment by seppestas** [comment]
cc @Alhadis

**Comment by Alhadis** [comment]
Alright. I'm the one who started this mess, so I should logically be the one to help with cleaning up.

First of all, thank you sincerely for stepping in and lending your knowledge here. =) I think it's pretty clear from #3743 that my knowledge of PCB-related technologies is at utmost zero, and I found working with PCB programs to be a frustrating and awkward affair. Had I known there were that many things unaccounted for, I surely would've bugged the owners of the projects I sourced samples from for their assistance.

Secondly, it was mentioned several times in #3784 that there were still some project files being unaccounted for. Which of a KiCad project's files have yet to be recognised by Linguist?

**Comment by Alhadis** [comment]
@seppestas As an aside, it would also be nice to have educated feedback on the [grammar-package](https://github.com/Alhadis/language-pcb) I wrote to provide syntax highlighting for PCB formats... ;)

**Comment by seppestas** [comment]
This PR just fixes an issue introduced in #3743 that makes it harder to find KiCad projects, namely confusion between "KiCad Boards" and "KiCad Layouts" and the fact that the barely used and deprecated "KiCad board" layout is listed before the "KiCad Layout" format in the advanced search list.

I think the further separation of KiCad library / project files requires some more discussion, and I don't plan to add these kind of changes in this PR.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2170,6 +2163,13 @@ KiCad Layout:
...
   codemirror_mime_type: text/x-common-lisp
   language_id: 187
+KiCad Legacy Layout:
+  type: data
+  extensions:
+  - ".brd"
+  tm_scope: source.pcb.board
+  ace_mode: text
+  language_id: 140848857
 KiCad Schematic:
```
Comment (Alhadis) [inline lib/linguist/languages.yml]: Are schematics still in use by KiCad projects? From what I saw, they featured a very similar syntax to the legacy `.brd` files.

Comment (5N44P) [reply lib/linguist/languages.yml]: Are you referring to `.sch` files? 
Yes, those are not deprecated

**Comment by Alhadis** [comment]
There was another file format I noticed that appeared related to KiCad projects... it had an extension of `.drl` and looked quite similar to the `.brd` file format, IIRC. [Here's one example](https://github.com/r2aiv/motor_driver/blob/eaa5e3c2f745a467b6b55ff642caf41fba41be7c/Gerber/pwm_motor_ctrl_2_order/pwm_motor_ctrl_2.drl).

Should this be included as a KiCad-related file too?

**Comment by 5N44P** [comment]
`.drl` is an extension used for Gerber drill files. Gerbers are the main fabbrication output for pcbs, and they are kind of standard. (some EDAs export `txt` files instead of `drl`). 

Drill files should contain, for each drill hole in the pcb, the coordinates and the tool number. Sometimes there's also a table correlating tool number with the hole diameter, so the manufacturer knows which drill bit sizes load into the machine. 

As an example: when you're done with the layout of your board, you generate your gerber and drill files, pack them in a `zip` folder and send it to the manufacturer. 

So `.drl` it's a generic Gerber file, not specific to KiCad. If it has to be classified, l'd put it in a "Gerber" language (which I don't know if it exists yet) 

If you have any pcb/KiCad/EDA questions, please ask, i'd be glad to help :smile:

**Comment by Alhadis** [comment]
Glad to know I have somebody to turn to, finally! =) Thank you. 😄 

> If it has to be classified, l'd put it in a "Gerber" language (which I don't know if it exists yet)
Already taken care of; see #3744. ;) And I now recall it was when I was researching Gerber images that I encountered the `.drl` extension and wondered what it was for... I left it out due to uncertainty. Though I should probably think about adding it back, haha.

File: lib/linguist/languages.yml (hunk 1 above)
Comment (seppestas) [reply lib/linguist/languages.yml]: The plan is to switch to an s-expression based syntax similar to the one used in the new KiCad layout in KiCad 5 though (but I think the plan is to change the file extension as well when doing this).

**Comment by seppestas** [comment]
Oops, looks like I accidently pushed my commit to my fork's master branch -.- Forgot about Github automatically creating patch branches.

**Comment by seppestas** [comment]
All issues should be fixed now. I think squashing the commits is probably a good idea :).

**Comment by Alhadis** [comment]
> I think squashing the commits is probably a good idea :)

Don't worry, we always squash-merge pull-requests. ;) It makes for a much tidier revision history.

Thank you!

/cc @lildude

**Comment by seppestas** [comment]
🎉

**Pull request merged**